tests/components/tado/test_config_flow.py

```diff
@@ -260,3 +260,120 @@ async def test_form_homekit(hass: HomeAssistant) -> None:
         ),
     )
     assert result["type"] == "abort"
+
+
+async def test_import_step(hass: HomeAssistant) -> None:
+    """Test import step."""
+    mock_tado_api = _get_mock_tado_api(getMe={"homes": [{"id": 1, "name": "myhome"}]})
+
+    with patch(
+        "homeassistant.components.tado.config_flow.Tado",
+        return_value=mock_tado_api,
+    ), patch(
+        "homeassistant.components.tado.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                "username": "test-username",
+                "password": "test-password",
+                "home_id": 1,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["data"] == {
+        "username": "test-username",
+        "password": "test-password",
+        "home_id": "1",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_step_existing_entry(hass: HomeAssistant) -> None:
+    """Test import step with existing entry."""
+    with patch(
+        "homeassistant.components.tado.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        entry = MockConfigEntry(
```

Comment:
Move the mock config entry creation and addition to the top. It doesn't need to be within the context manager.